lib/elixir/test/elixir/module/types/integration_test.exs

```diff
@@ -483,7 +486,7 @@ defmodule Module.Types.IntegrationTest do
                 dynamic(
                   %Date{} or %DateTime{} or %NaiveDateTime{} or %Time{} or %URI{} or %Version{} or
                     %Version.Requirement{}
-                ) or atom() or binary() or float() or integer() or list(term())
+                ) or atom() or binary() or empty_list() or float() or integer() or non_empty_list(term(), term())
```

Comment:
I wonder if we should format this as `maybe_improper_list(term(), term())` instead (and make a public helper for it in descr)